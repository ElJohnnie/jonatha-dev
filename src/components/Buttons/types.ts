export interface AnimateButtonProps {
  route: string;
  title: string;
  icon?: JSX.Element;
}
