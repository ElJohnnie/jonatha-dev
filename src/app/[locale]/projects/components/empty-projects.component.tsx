export default function EmptyBlog({ text }: Readonly<{ text: string }>) {
  return (
    <section className='flex h-full flex-col items-center justify-center'>
      <div className='mx-auto flex text-center lg:max-w-none'>
        <h1 className='mb-4 text-3xl font-extrabold leading-none tracking-tight md:text-6xl dark:text-white'>
          {text}
        </h1>
      </div>
    </section>
  );
}
