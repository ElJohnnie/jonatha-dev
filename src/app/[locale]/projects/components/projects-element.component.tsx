import { ProjectElementComponent } from '../types';
import Link from 'next/link';
import Image from 'next/image';
import { useLocale } from 'next-intl';

export default function ProjectElement({
  project,
}: Readonly<ProjectElementComponent>) {
  const locale = useLocale();
  return (
    <>
      <article
        className='group relative h-64 w-full overflow-hidden rounded-lg bg-cover bg-center shadow-lg transition  duration-300 ease-in-out hover:shadow-2xl'
        style={{
          backgroundImage:
            "url('https://images.unsplash.com/photo-1623479322729-28b25c16b011?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=1740&q=80')",
        }}
      >
        <div className='absolute inset-0 bg-black bg-opacity-50 transition duration-300 ease-in-out group-hover:opacity-75'></div>
        <div className='relative flex h-full w-full items-center justify-center px-4 sm:px-6 lg:px-4'>
          <h3 className='text-center'>
            <Link
              className='text-center text-2xl font-bold text-white'
              href={`/${locale}/projects/${project.slug}`}
            >
              <span className='absolute inset-0'></span>
              {project.title}
            </Link>
          </h3>
        </div>
      </article>
    </>
  );
}
