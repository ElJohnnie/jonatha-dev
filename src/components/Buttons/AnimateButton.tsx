import Link from 'next/link';
import { AnimateButtonProps } from './types';

export default function AnimateButton({
  route,
  title,
  icon,
}: Readonly<AnimateButtonProps>) {
  return (
    <Link
      href={route}
      className='align-center group relative inline-flex items-center justify-center overflow-hidden rounded border border-transparent bg-transparent px-12 py-3 text-sm font-bold text-slate-100 hover:text-indigo-500 focus:outline-none focus:ring active:bg-indigo-600 active:text-white'
    >
      <span className='ease absolute left-0 top-0 h-0 w-0 border-t-2 border-indigo-500 transition-all duration-200 group-hover:w-full'></span>
      <span className='ease absolute right-0 top-0 h-0 w-0 border-r-2 border-indigo-500 transition-all duration-200 group-hover:h-full'></span>
      <span className='ease absolute bottom-0 right-0 h-0 w-0 border-b-2 border-indigo-500 transition-all duration-200 group-hover:w-full'></span>
      <span className='ease absolute bottom-0 left-0 h-0 w-0 border-l-2 border-indigo-500 transition-all duration-200 group-hover:h-full'></span>
      <span className='flex items-center justify-center align-middle'>
        {title}
        {icon && <span className='ml-2'>{icon}</span>}
      </span>
    </Link>
  );
}
