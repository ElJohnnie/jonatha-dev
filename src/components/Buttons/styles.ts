const linkStyle = [
  'align-center',
  'group',
  'relative',
  'inline-flex',
  'items-center',
  'justify-center',
  'overflow-hidden',
  'rounded',
  'border',
  'border-transparent',
  'bg-transparent',
  'px-12',
  'py-3',
  'text-sm',
  'font-bold',
  'text-slate-200/60',
  'hover:text-indigo-500',
  'focus:outline-none',
  'focus:ring',
  'active:bg-indigo-600',
  'active:text-white',
];

const linkStyleString = linkStyle.join(' ');

export { linkStyleString };
