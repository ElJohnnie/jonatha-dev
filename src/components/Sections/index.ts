import Hero from './Hero/Hero';
import About from './About/About';
import Blog from './Blog/Blog';

export { Hero, About, Blog };
