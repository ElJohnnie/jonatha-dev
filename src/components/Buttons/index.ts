import AnimateButton from './AnimateButton';

export { AnimateButton };
