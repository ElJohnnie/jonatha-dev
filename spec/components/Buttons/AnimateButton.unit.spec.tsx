import { AnimateButton } from '@/components/Buttons';
import { render } from '@testing-library/react';

const props = {
  route: '/',
  title: 'any',
};

describe('should render correctly AnimateButton component', () => {
  it('match snapshot', () => {
    const { container } = render(<AnimateButton {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
