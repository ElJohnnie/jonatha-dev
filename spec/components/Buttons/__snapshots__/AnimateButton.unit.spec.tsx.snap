// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly AnimateButton component match snapshot 1`] = `
<a
  class="align-center group relative inline-flex items-center justify-center overflow-hidden rounded rounded-md border border-transparent bg-transparent px-12 py-3 text-sm font-bold text-slate-100 hover:text-indigo-500 focus:outline-none focus:ring active:bg-indigo-600 active:text-white"
  href="/"
>
  <span
    class="ease absolute left-0 top-0 h-0 w-0 rounded-md border-t-2 border-indigo-500 transition-all duration-200 group-hover:w-full"
  />
  <span
    class="ease absolute right-0 top-0 h-0 w-0 rounded-md border-r-2 border-indigo-500 transition-all duration-200 group-hover:h-full"
  />
  <span
    class="ease absolute bottom-0 right-0 h-0 w-0 rounded-md border-b-2 border-indigo-500 transition-all duration-200 group-hover:w-full"
  />
  <span
    class="ease absolute bottom-0 left-0 h-0 w-0 rounded-md border-l-2 border-indigo-500 transition-all duration-200 group-hover:h-full"
  />
  <span
    class="flex items-center justify-center align-middle"
  >
    any
  </span>
</a>
`;
